lab-bundle-spec.md

```diff
@@ -308,20 +308,140 @@ environment:
       reference: primary_user.password
 ```
 
-### Activity Tracking (Alpha)
+### Activity Tracking
 
 Activity tracking is a feature for evaluating a student's performance in a lab by running a script at "checkpoints". These scripts can call APIs relevant to any environment resource to query their current state. For example, the script may inspect and validate the configuration of GCE instances running in `my-project`, to ensure the user is following the instructions properly.
 
-Lab bundles will provisionally support the JSON representation of Activity Tracking currently used in the Qwiklabs web interface. The JSON definition should be stored in file separately from (and referenced directly in) `qwiklabs.yaml`.
+A lab has an Assessment, which in turn contains the Steps (checkpoints).
 
-```yaml
-entity_type: Lab
+#### Assessment
 
-...
+attribute          | required | type    | notes
+-------------------| -------- | --------| --------------------------------------
+passing_percentage | ✓        | integer | The percentage of total points the student must achieve to "pass" the lab.
+steps              | ✓        | array   | An array of [Steps](#steps)
 
-activity_tracking: ./assessment/activity_tracking.json
+```yml
+assessment:
+  passing_percentage: 75
+  steps: ...
 ```
 
-> **Note:** Support for this format should be considered deprecated.
->
-> Further work will be done to define a new DSL for expressing Activity Tracking logic. Exploratory sketches of this DSL can be found in the `./examples` directory of this repository.
+#### Steps
+
+attribute        | required | type                       | notes
+-----------------| -------- | ---------------------------| --------------------------------------
+title            | ✓        | locale dictionary          |
+maximum_score    | ✓        | integer                    | The maximum number of points this step can award.
+student_messages | ✓        | array of objects           | The keys are how the messages will be referenced in the code, and the values are locale dictionaries.
+services         | ✓        | array of resource services | An array of services that will be used in the code block. Each resource type specifies a set of allowed services.
+code             | ✓        | string                     | Code to be executed. See [below](#code) for more information.
+
+```yml
+assessment:
+  passing_percentage: 75
+  steps:
+    - title:
+        locales:
+          en: Create a Cloud Storage bucket
+          es: Crear un depósito de almacenamiento en la nube
+      maximum_score: 5
```

Comment:
I know you removed the pass/fail checkbox based on its lack of usage but I'm curious how many steps offer partial points.

Instead of only allowing integer scores and passing percentages, I'd like to propose only allowing pass/fail steps and a passing step percentage (which defaults to all of them in Git Whisperer).

Reply:
it's not just a question of how many steps offer partial points - this would also change things if a given assessment has steps with different `max`s. it would also change how game points are calculated. i think i'd rather leave as-is for now, unless others feel strongly.

Reply:
Is there a reason the key is `maximum_score`, but in the code example below we pass in `max_score`?

Reply:
in the code example it used to be a positional param so it didn't matter, but consistency would probably have been nice. it's now a keyword param and therefore does matter, and has been updated accordingly.